ambry-cloud/src/main/java/com.github.ambry.cloud/azure/CosmosDataAccessor.java

```diff
@@ -194,6 +235,43 @@ private CloudBlobMetadata createMetadataFromDocument(Document document) {
     return resourceResponse;
   }
 
+  /**
+   * Utility method to call Cosmos document query method and record the query time.
+   * @param sqlQuerySpec the DocumentDB query to execute.
+   * @param feedOptions {@link FeedOptions} object specifying the options associated with the method.
+   * @param timer the {@link Timer} to use to record query time (excluding waiting).
+   * @return {@link BlockingObservable} object containing the query response.
+   */
+  private BlockingObservable<FeedResponse<Document>> executeCosmosQuery(SqlQuerySpec sqlQuerySpec,
+      FeedOptions feedOptions, Timer timer) {
+    Timer.Context operationTimer = timer.time();
+    try {
+      return asyncDocumentClient.queryDocuments(cosmosCollectionLink, sqlQuerySpec, feedOptions).toBlocking();
+    } finally {
+      operationTimer.stop();
+    }
+  }
+
+  /**
+   * Utility method to call Cosmos change feed query method and record the query time.
+   * @param changeFeedOptions {@link ChangeFeedOptions} object specifying the options associated with the method.
+   * @param timer the {@link Timer} to use to record query time (excluding waiting).
+   * @return {@link FeedResponse} object representing the query response.
+   */
+  private FeedResponse<Document> executeCosmosChangeFeedQuery(ChangeFeedOptions changeFeedOptions, Timer timer) {
+    Timer.Context operationTimer = timer.time();
+    try {
+      // FIXME: Using single() for the observable returned by toBlocking() works for now. But if a high enough maxFeedSize
```

Comment:
to address this, could you use `toIterable()` and have this method return `Iterable<FeedResponse<Document>>`?

Reply:
I have a ticket filed for this here -- https://jira01.corp.linkedin.com:8443/browse/AMBRY-6817 (added to backlog and blueshift epic and assigned to me).
Will fix this with this ticket. Have added this note in the ticket.